docs/testing/e2e-framework-design-dimensions.md

```diff
@@ -0,0 +1,396 @@
+
+### Self- Versus Remotely-managed
+
+|                      |          |
+|-------------------- |-------- |
+| Type                 | string   |
+| Domain               | discreet |
+| Cardinality estimate | 2        |
+
+Self-managed clusters are a good candidate for later implementation.
+
+
+<a id="org1e1448b"></a>
+
+### Bundle Version
+
+|                      |                                                                                         |
+|-------------------- |--------------------------------------------------------------------------------------- |
+| Type                 | string                                                                                  |
+| Domain               | discreet<sup><a id="fnr.2" class="footref" href="#fn.2" role="doc-backlink">2</a></sup> |
+| Cardinality estimate | 2<sup><a id="fnr.3" class="footref" href="#fn.3" role="doc-backlink">3</a></sup>        |
+
+Testing packages against an older bundle doesn't make sense. But it could be useful for testing that the controller can deal with changes in the format of the bundle itself over time.
+
+
+<a id="orge08c449"></a>
+
+## Worst-case Cardinality
+
+The maximum number of test dimension permutations is likely to be greater than that which can be run in a reasonable amount of time. Below is a worst-case scenario:
+
+|           |                              |
+|--------- |---------------------------- |
+| 5         | Providers                    |
+| 5         | Provider versions            |
+| 5         | CLI versions                 |
+| 5         | Kubernetes versions          |
+| 10        | Node OSes                    |
+| 5         | Node OS versions             |
+| 5         | Packages controller versions |
+| 5         | Package versions             |
+| 2         | Architectures                |
+| 2         | Self vs Remotely managed     |
+| 2         | Bundle version               |
+| 6,250,000 | Total permutations           |
+
+With 6,250,000 permutations, running on 32 CPUs at a rate of one test per CPU per minute, a complete test run would take ≈ 4½ months.
+
+A more middle-of-the-road estimate might be:
+
+|       |                              |
+|-------|------------------------------|
+| 5     | Providers                    |
+| 1.6   | Provider versions            |
+| 2     | CLI versions                 |
+| 4     | Kubernetes versions          |
+| 3     | Node OSes                    |
+| 2     | Node OS versions             |
+| 2     | Packages controller versions |
+| 2     | Package versions             |
+| 1     | Architectures                |
+| 1     | Self vs Remotely managed     |
+| 1     | Bundle version               |
+| 1,536 | Total permutations           |
+
+With this far more tractable scenario, a complete run would take forty-eight minutes. Of course this still assumes 32 CPUs available and able to run any test (none of which take more than a minute from beginning to end).
```

Comment:
This 48 min value is per package right?

Reply:
It would also be per test flow correct? For example if we there was a e2e test scenario for harbor to make sure images are imported correctly, would that need to be tested through all the permutations?

Reply:
Yes, per test.